roles/tas_single_node/tasks/certificates.yml

```diff
@@ -101,18 +106,30 @@
           -in '{{ tas_single_node_remote_fulcio_private_key }}'
           -out '{{ tas_single_node_remote_fulcio_public_key }}'
           -passin 'pass:{{ tas_single_node_fulcio_ca_passphrase }}'
-        creates: "{{ tas_single_node_remote_fulcio_public_key }}"
+      changed_when: >
+        (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_remote_fulcio_public_key) | list | length) == 0
+      when: >
+        tas_single_node_fulcio_private_key != ""
+        or (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_remote_fulcio_public_key) | list | length) == 0
```

Comment:
* Will this command work even if user provides an RSA key?
* For RSA keys, generating the public key can yield a different output every time. I think what we should do is we should have a task that finds out whether the public key corresponds to the private key; if not, then this task should regenerate.